src/topology/homotopy/homotopy_group.lean

```diff
@@ -0,0 +1,209 @@
+/-
+Copyright (c) 2021 Roberto Alvarez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Roberto Alvarez
+-/
+
+import topology.homotopy.basic
+import topology.path_connected
+import topology.unit_interval
+import algebraic_topology.fundamental_groupoid.fundamental_group
+
+/-!
+# `n`th homotopy group
+
+We define the `n`th homotopy group at `x`, `π n x`, as the equivalence classes
+of functions from the nth dimensional cube to the topological space `X`
+that send the boundary to the base point `x`, up to homotopic equivalence.
+Note that such functions are generalized loops `gen_loop n x`, in particular
+`gen_loop 1 x ≃ path x x`
+
+We show that `π 0 x` is equivalent to the path-conected components, and
+that `π 1 x` is equivalent to the fundamental group at `x`.
+
+## definitions
+
+* `gen_loop n x` is the type of continous fuctions `I^n → X` that send the boundary to `x`
+* `homotopy_group n x` denoted `π n x` is the quotient of `gen_loop n x` by homotopy relative
+  to the boundary
+
+TODO: show that `π n x` is a group when `n > 0` and abelian when `n > 1`. Show that
+`pi1_equiv_fundamental_group` is an isomorphism of groups.
+
+-/
+
+open_locale unit_interval topological_space
+
+noncomputable theory
+
+universes u
+variables {X : Type u} [topological_space X]
+variables {n : ℕ} {x : X}
+
+/--
+The `n`-dimensional cube.
+-/
+@[derive [has_zero, has_one, topological_space]]
+def cube (n : ℕ) : Type := fin n → I
+notation `I^` := cube
+
+namespace cube
+
+@[continuity] lemma proj_continuous {n} (i : fin n) : continuous (λ f : I^n, f i) :=
+begin
+  apply continuous_infi_dom,
+  rw continuous_def,
+  exact λ s H, is_open_induced_iff'.2 ⟨s, H, rfl⟩
+end
```

Comment:
```suggestion
continuous_apply i
```

Reply:
Excellent! That proof is essentially refl but I didn't knew how to golf it